src/pubsub/src/publisher/publisher.rs

```diff
@@ -103,16 +109,31 @@ impl PublisherBuilder {
     }
 
     pub fn build(self) -> Publisher {
+        // Enforce limits by clamping the user-provided options.
+        let batching_options = BatchingOptions::new()
+            .set_delay_threshold(std::cmp::min(
+                self.batching_options.delay_threshold,
+                MAX_DELAY,
+            ))
```

Comment:
Does this work? Is it any better?


```suggestion
            .set_delay_threshold(
                self.batching_options.delay_threshold.clamp(Duration::ZERO,
                MAX_DELAY,
            ))
```

Reply:
Yeah that also works! had that in a previous iteration of the PR. If durations can be negative we definitely want this.